toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -247,8 +250,24 @@ func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string,
 	)
 	defer func() {
 		if err != nil {
-			span.RecordError(fmt.Errorf("image customization failed"))
-			span.SetStatus(codes.Error, "image customization failed")
+			category := GetErrorCategory(err)
+			code := GetErrorCode(err)
+			
+			// Use GetErrorJSON to create JSON representation without PII
+			span.RecordError(fmt.Errorf("%s", GetErrorJSON(err)))
+			
+			// Set span attributes
+			span.SetAttributes(
```

Comment:
actually, we don't need to set the category and code as span attributes, lets remove this part of the logic. we only need to call RecordError() and SetStatus()